code/common/cloudutils.q

```diff
@@ -0,0 +1,41 @@
+/a function to execute system commands and return a log message depending on the resulting exit code:
+/used for both launchprocess.sh and killprocess.sh
+syscomm:{[params;cmd]
+    /params is (i) a dictionary if syscomm has been called by launch, and (ii) a string if it has been called by killproc
+    /cmd is the command to be executed, as a string
+    
+    prevexitcode:first "I"$system cmd,"; echo $?"; 
+    $[lok:99h=type params;[id:`launchprocess;pname:params[`procname]];[id:`killprocess;pname:params]];
```

Comment:
I think it's a little clearer to do this as two conditionals:

```suggestion
    id:$[lok:99h=type params;`launchprocess;`killprocess];
    pname:$[lok;params`procname;params];
```

IMO the code-block-within-$ syntax gets confusing very quickly and this just makes it a little clearer at a glance where these two variables are being defined

Reply:
Can you have another look at this comment?